### PR TITLE
ECAL esConsumes migrations in EventFilter and RecoLocalCalo

### PR DESCRIPTION
**Pull request by thomreis**
#### PR description:

esConsumes migrations of remaining `EventSetupRecord::get` occurrences in EventFilter/EcalRawToDigi and RecoLocalCalo/EcalRecProducers.

#### PR validation:

Complies and passes 136.874.

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by sroychow** [comment]
@thomreis I have marked the google [doc](https://docs.google.com/spreadsheets/d/1cUypk2EK4xVcEpFGtaNqXrWZEoFz_E9l4gNN3YoBQH4/edit?usp=sharing) with this PR info. In future  if you are migrating other modules, please update the doc.

**Comment by thomreis** [comment]
Thanks @sroychow I will update the doc for future migrations.

File: RecoLocalCalo/EcalRecProducers/plugins/EcalRecHitProducerGPU.cc [hunk 1]
```diff
@@ -177,15 +198,15 @@ void EcalRecHitProducerGPU::acquire(edm::Event const& event,
...
+  IntercalibConstantsHandle_ = setup.getHandle(tokenIntercalibConstants_);
+  ChannelStatusHandle_ = setup.getHandle(tokenChannelStatus_);
+
+  LaserAPDPNRatiosHandle_ = setup.getHandle(tokenLaserAPDPNRatios_);
+  LaserAPDPNRatiosRefHandle_ = setup.getHandle(tokenLaserAPDPNRatiosRef_);
+  LaserAlphasHandle_ = setup.getHandle(tokenLaserAlphas_);
+  LinearCorrectionsHandle_ = setup.getHandle(tokenLinearCorrections_);
+  recHitParametersHandle_ = setup.getHandle(tokenRecHitParameters_);
 
   auto const& ADCToGeVConstantProduct = ADCToGeVConstantHandle_->getProduct(ctx.stream());
```
Comment (slava77) [inline RecoLocalCalo/EcalRecProducers/plugins/EcalRecHitProducerGPU.cc:211]: are the handles actually needed? It seems like their usefulness is only as a temporary to be immediately dereferenced.
`getData` here seems more appropriate.
```c++
   auto const& adcToGeVConstantProduct = setup.getData(tokenADCToGeVConstant_).getProduct(ctx.stream());
```

please check and cleanup.

Comment (thomreis) [reply RecoLocalCalo/EcalRecProducers/plugins/EcalRecHitProducerGPU.cc:211]: Only two of those are needed. I will replace the rest with `getData()`.

**Comment by perrotta** [comment]
please test

File: RecoLocalCalo/EcalRecProducers/plugins/EcalRecHitProducerGPU.cc (hunk 1 above)
Comment (slava77) [reply RecoLocalCalo/EcalRecProducers/plugins/EcalRecHitProducerGPU.cc]: > Only two of those are needed. I will replace the rest with `getData()`.

thank you.

the remaining two are used only locally and could as well be defined just in the `::acquire` method itself.

**Comment by slava77** [comment]
+reconstruction

for https://github.com/cms-sw/cmssw/pull/34273	2763d63 
- code changes are in line with the PR description and (mostly) with the follow up review
- jenkins tests pass and comparisons with the baseline show no (relevant) differences

**Comment by qliphy** [comment]
+1

**Pull request merged**